prow/plugins/reopen/reopen.go

```diff
@@ -74,9 +75,19 @@ func handle(gc githubClient, log *logrus.Entry, e *github.GenericCommentEvent) e
 
 	if e.IsPR {
 		log.Info("Re-opening PR.")
-		return gc.ReopenPR(org, repo, number)
+		err := gc.ReopenPR(org, repo, number)
+		if err != nil {
+			resp := fmt.Sprintf("failed to re-open PR: %s", err)
```

Comment:
As is, the error message looks like:
```
"status code 422 not one of [200], body: {"message":"Validation Failed","errors":[{"resource":"PullRequest","code":"custom","field":"state","message":"state cannot be changed. The bump_kubernetes_to_1.8 branch was force-pushed or recreated."}],"documentation_url":"https://developer.github.com/v3/pulls/#update-a-pull-request"}"
```
which isn't formatted very nicely for a github comment.
If we are going to include the error text in the comment we might want to specifically handle the 422 error in `ReopenPR` and `ReopenIssue` and return a better message in that case.

I think a better alternative to using the error text is to change the signature of `ReopenPR` and `ReopenIssue` to return `(bool, error)` where the bool indicates whether the reopen was successful. A status of 422 would be treated as not an error, and `(false, nil)` would be returned indicating that the reopen failed and we need to leave a comment. This would distinguish the 'acceptable' error case from the other error cases which would return `(false, <non-nil-error>)` and should not result in comments being created.

Reply:
Yeah I agree that the formatting right now would not be ideal but are we just not going to comment if the re-open fails for some other reason?

Reply:
That would work. I think it's a bit more idiomatic to return just `error` but use a special error type that can be checked by the user. We do this for [assign and unassign](https://github.com/kubernetes/test-infra/blob/62ab514cbd6330af87c505c6cd1ccaf48345e8c8/prow/github/client.go#L613-L685) and it works well.

Reply:
> Yeah I agree that the formatting right now would not be ideal but are we just not going to comment if the re-open fails for some other reason?

There is a general policy in prow to only comment for "expected" failures, whatever that means. For unexpected failures we should just log it, and maybe fire an alert if there are lots of them. It might be worth revisiting this policy.

Reply:
Implementing the custom error type, which is definitely more idiomatic.

Reply:
I'm extracting the actual "state cannot be changed. ..." message now